OracleDatabase/12.1.0.2/Vagrantfile

```diff
@@ -20,38 +20,91 @@ VAGRANTFILE_API_VERSION = "2"
 BOX_URL = "https://oracle.github.io/vagrant-boxes/boxes"
 BOX_NAME = "oraclelinux/7"
 
-# define hostname
-NAME = "oracle-12102-vagrant"
-
 unless Vagrant.has_plugin?("vagrant-proxyconf")
   puts 'Installing vagrant-proxyconf Plugin...'
   system('vagrant plugin install vagrant-proxyconf')
 end
 
-# get host time zone for setting VM time zone, if possible
-# can override in env section below
-offset_sec = Time.now.gmt_offset
-if (offset_sec % (60 * 60)) == 0
-  offset_hr = ((offset_sec / 60) / 60)
-  timezone_suffix = offset_hr >= 0 ? "-#{offset_hr.to_s}" : "+#{(-offset_hr).to_s}"
-  SYSTEM_TIMEZONE = 'Etc/GMT' + timezone_suffix
-else
-  # if host time zone isn't an integer hour offset, fall back to UTC
-  SYSTEM_TIMEZONE = 'UTC'
+# Define constants
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  # Use vagrant-env plugin if available
+  if Vagrant.has_plugin?("vagrant-env")
+    config.env.load(".env.local", ".env") # enable the plugin
+  end
+
+  # VM name
+  NAME = default_s('NAME', 'oracle-12102-vagrant')
+
+  # Memory for the VM (in MB, 2048 MB = 2 GB)
+  MEMORY = default_i('MEMORY', 2048)
+
+  # VM time zone
+  # If not specified, will be set to match host time zone (if possible)
+  SYSTEM_TIMEZONE = default_s('SYSTEM_TIMEZONE', host_tz)
+
+  # Oracle base directory
+  # "VM_" prefix used to avoid conflict with host environment variable
+  VM_ORACLE_BASE = default_s('VM_ORACLE_BASE', '/opt/oracle')
```

Comment:
Why do we need the `VM_` prefix for this? Would this overwrite or collide with already used variables on the host?

Reply:
If one does a lot of Oracle database work, then it's very possible that the ORACLE_BASE, ORACLE_HOME and ORACLE_SID environment variables are set on their host. (They are on mine.) Those environment variables are almost certainly not what we want in the VM, especially if the host is Windows. Using the `VM_` prefix should prevent conflicts.

Reply:
Ok, thank you, that makes sense. I suppose that would bring us back to the question, or similar question, that @Djelibeybi raised. Should we then not prefix more environment variables? For example, `SYSTEM_TIMEZONE` is opaque as to whether it's the host system or the guest system, so it's probably better to prefix that one as well with `VM_`.

Same with the Oracle related environment variables.  It's equally conceivable that a user has `ORACLE_PDB` also set on his local machine, or `ORACLE_CHARACTERSET`.

Would it not be less confusing prefixing all Vagrant related environment variables with `VM_` then?

Note, that does not mean that we should pass them into the install script as such. There is a benefit not polluting the install script with `VM_` as it can be reused for other tasks. We should still apply a mapping of the "host" `VM_` prefixed environment variables to the actual environment variables within the box, i.e:

```
  config.vm.provision "shell", path: "scripts/install.sh", env:
    {
       "SYSTEM_TIMEZONE"     => VM_SYSTEM_TIMEZONE,
       "ORACLE_BASE"         => VM_ORACLE_BASE,
       "ORACLE_HOME"         => VM_ORACLE_HOME,
       "ORACLE_SID"          => VM_ORACLE_SID,
       "ORACLE_PDB"          => VM_ORACLE_PWD
    }
```